bloom_nofos/users/auth/login_gov.py

```diff
@@ -0,0 +1,159 @@
+                    (key for key in keys["keys"] if key.get("kty") == "RSA"), None
+                )
+                if not jwk:
+                    raise ValueError("No RSA key found in Login.gov certs")
+
+            return RSAAlgorithm.from_jwk(json.dumps(jwk))
+
+        except Exception as e:
+            raise ValueError(f"Error fetching Login.gov public key: {str(e)}")
+
+    def get_authorization_url(self, state=None):
+        """Generate the Login.gov authorization URL for the initial redirect."""
+        state = state or str(uuid.uuid4())
+        nonce = str(uuid.uuid4())
+
+        params = {
+            "client_id": self.client_id,
+            "response_type": "code",
+            "acr_values": self.acr_values,
+            "scope": "openid email",
+            "redirect_uri": self.redirect_uri,
+            "state": state,
+            "nonce": nonce,
+        }
+
+        auth_url = f"{self.oidc_url}/authorize?{urlencode(params)}"
+        return auth_url, state, nonce
+
+    def get_token(self, code):
+        """Exchange authorization code for tokens."""
+        if not self.private_key:
+            raise Exception(
+                "Private key not configured. Cannot create client assertion."
+            )
+
+        now = int(time.time())
+
+        try:
+            private_key = load_pem_private_key(
+                self.private_key, password=None, backend=default_backend()
+            )
+
+            assertion = jwt.encode(
+                {
+                    "iss": self.client_id,
+                    "sub": self.client_id,
+                    "aud": f"{self.token_url}/token",
+                    "jti": str(uuid.uuid4()),
+                    "exp": now + 300,  # 5 minutes
+                    "iat": now,
+                },
+                private_key,
+                algorithm="RS256",
+            )
+        except Exception as e:
+            raise Exception(f"Error creating client assertion: {str(e)}")
+
+        token_data = {
+            "grant_type": "authorization_code",
+            "code": code,
+            "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+            "client_assertion": assertion,
+        }
+
+        response = requests.post(f"{self.token_url}/token", data=token_data)
+        response.raise_for_status()
+        return response.json()
+
+    def validate_id_token(self, id_token, nonce):
+        """Validate the ID token from Login.gov."""
+        try:
+            unverified_header = jwt.get_unverified_header(id_token)
+            kid = unverified_header.get("kid")
+
+            public_key = self._get_login_gov_public_key(kid)
+            if not public_key:
+                raise ValueError("Could not get Login.gov public key")
+
+            decoded = jwt.decode(
+                id_token,
+                public_key,
+                algorithms=["RS256"],
+                audience=self.client_id,
+            )
```

Comment:
Should we also add `issuer="https://google.com",` so that we make sure these requests come from the right login.gov url?

This should be the same value as `LOGIN_GOV_OIDC_URL`.

Reply:
Yes, good call. added this